NEWS.md

```diff
@@ -170,6 +170,10 @@ Deprecated or removed
 External dependencies
 ---------------------
 
+- The terminal info database, `terminfo`, is now vendored by default, providing a better
+  REPL user experience when `terminfo` is not available on the system. Julia can be built
+  without vendoring the database using the Makefile option `WITH_TERMINFO=0`. ([#55411])
```

Comment:
Any reason why you didn't call this `USE_SYSTEM_TERMINFO` for consistency with other variables?

Reply:
I believe I had a reason at the time because I do remember thinking about that, though I don't recall what that reason was at this point...

I guess one difference is that `USE_SYSTEM_*` implies different behavior based on `USE_BINARYBUILDER`, which is not the case here. There are other variables called `WITH_*`, though I can't recall offhand what they are.

Reply:
It would also be a bit of a misnomer since we actually _do_ still prefer an existing terminfo on the system if available, this only controls whether we bundle terminfo to fall back on if the system doesn't have one

Reply:
OK yeah it's a bit different from other variables.